packages/wrangler/src/deploy/deploy.ts

```diff
@@ -961,16 +969,23 @@ See https://developers.cloudflare.com/workers/platform/compatibility-dates for m
 							})
 						)
 					);
-				}
 
-				// Update service and environment tags when using environments
-				if (hasDefinedEnvironments(config)) {
-					await applyServiceAndEnvironmentTags(
-						config,
-						accountId,
-						scriptName,
-						tags
-					);
+					// Update service and environment tags when using environments
+					const nextTags = applyServiceAndEnvironmentTags(config, tags);
+					if (tags !== nextTags) {
+						try {
+							await patchNonVersionedScriptSettings(
+								props.config,
+								accountId,
+								scriptName,
+								{
+									tags: nextTags,
+								}
+							);
+						} catch {
+							warnOnErrorUpdatingServiceAndEnvironmentTags();
+						}
+					}
```

Comment:
Is this needed now the tags are set above?

Reply:
this handles updating the tags for the code branch where the Worker is brand new, i.e. (`!workerExists` / `!canUseNewVersionsDeploymentsApi`)